### PR TITLE
v4.19.52: patch and config update for v4.19.52 on AArch64

### PR DESCRIPTION
**Pull request by Pennyzct**
Hi~ @egernst I have already done patch and config update for v4.19.52 on AArch64.

### CODE REVIEW

**Comment by Pennyzct** [comment]
@gnawux  @egernst soooooo desperately looking forward to bump to v5.0.x. ;)

File: kernel/configs/arm64_kata_kvm_4.19.x [hunk 1]
```diff
@@ -500,6 +502,7 @@ CONFIG_ACPI_TABLE_UPGRADE=y
 # CONFIG_ACPI_DEBUG is not set
 # CONFIG_ACPI_PCI_SLOT is not set
 CONFIG_ACPI_CONTAINER=y
+CONFIG_ACPI_HOTPLUG_MEMORY=y
```
Comment (grahamwhaley) [inline kernel/configs/arm64_kata_kvm_4.19.x]: I presume the HOTPLUG additions are deliberate. May want to add a note to the commit message, as they look like not just part of an update, but are an addition.

**Comment by gnawux** [comment]
I could understand your feeling 🤔 

I prefer to stick on 4.19 before next LTS kernel is released, however, I don't mind aarch64 have a different kernel version.

File: kernel/configs/arm64_kata_kvm_4.19.x (hunk 1 above)
Comment (Pennyzct) [reply kernel/configs/arm64_kata_kvm_4.19.x]: yep, you just got me😱. I may need to split that to another commit...🤠

Comment (gnawux) [reply kernel/configs/arm64_kata_kvm_4.19.x]: There are several entries added in this patch. Do you have any explanation here @Pennyzct?

**Comment by Pennyzct** [comment]
@gnawux just imaging keep refining this whole big patch. it's killing me. dose everyone on board for AArch64 has different kernel version?? @egernst @jodh-intel @devimc @grahamwhaley

**Comment by gnawux** [comment]
@Pennyzct 
>  dose everyone on board for AArch64 has different kernel version??

I think this idea is a double-sword for you. It may help you avoid pick huge patches from non-LTS kernels, on the other hand, it might make others less care of ARM arch by some means.

I can't tell if it is a better choice for you.

**Comment by gnawux** [comment]
@grahamwhaley could we set this PR as a dependency of CI of kata-containers/runtime#1817?

**Comment by Pennyzct** [comment]
@gnawux emmmmmmm, But I have memory hot-remove, DAX, cpu hotplug, kvm-ptp on the way. 😱🙃

**Comment by grahamwhaley** [comment]
> @gnawux @egernst soooooo desperately looking forward to bump to v5.0.x. ;)

Well, there was a PR for that https://github.com/kata-containers/runtime/pull/1504 , but it got closed very recently due to 'inactivity', which maybe was not the right course of action.
It got stalled as the [kernel fragment PR](https://github.com/kata-containers/packaging/pull/314) got stuck on:
 - it seemed to make the firecracker CI less stable
 - it really should have SRIOV support added in

Personally, I suspect the right path is to move to v5.x kernels with full config files, and carry on with the fragment support work in parallel - unless there is a focussed effort to land the fragment work rsn.

**Comment by gnawux** [comment]
> emmmmmmm, But I have memory hot-remove, DAX, cpu hotplug, kvm-ptp on the way. 😱🙃

@Pennyzct awesome, how about co-present a topic on OpenInfra Summit Shanghai this November on your excellent upstream work.

**Comment by jcvenegas** [comment]
@Pennyzct could you add a depens on https://github.com/kata-containers/runtime/pull/1817

**Comment by jcvenegas** [comment]
@egernst  could you add a depends-on https://github.com/kata-containers/packaging/pull/592 ?

**Comment by Pennyzct** [comment]
Hi~ @bergwolf @jcvenegas I have already added depends-on. ;)

**Comment by gnawux** [comment]
checked the failure in the ARM test of kata-containers/runtime#1817, it is caused by lacking `ptp0` device, however, looks the fix in kata-containers/osbuilder#265 won't be merged soon.

 Let's merge this and kata-containers/runtime#1817 once the essential CIs are passed because this is a security related fix.

**Comment by bergwolf** [comment]
CI failed at `ERROR: Please bump version in kernel/kata_config_version`. @Pennyzct please bump kata_config_version.

**Comment by Pennyzct** [comment]
Hi~ @bergwolf  while you are on children-happy hour, I have done the update. ;)

**Comment by Pennyzct** [comment]
Hi~ @grahamwhaley I'm too lazy to split a new one, I'm guilty😈, I just address HOTPLUG additions on this commit

**Comment by gnawux** [comment]
again

```
ERROR: Please bump version in kernel/kata_config_version
```

**Comment by gnawux** [comment]
finally passed, merge it

**Comment by bergwolf** [comment]
Thanks @Pennyzct!

**Comment by devimc** [comment]
@gnawux snap-ci was failing and you didn't care about it and merged this anyway!!!  :-1: 

**I can't release the new snap because it's broken**, *Thanks @gnawux nice job!!* :-1:

**Comment by jodh-intel** [comment]
@devimc, @gnawux - Relax guys  :sunglasses: Mistakes happen. Nobody got hurt right?

I think we need to all focus on fixing the CI's so that a failure is a very unusual occurrence (/me looks sideways at https://github.com/kata-containers/osbuilder/pull/265 as that is causing constant failures .... ;)

But for now, can we take a deep breath, count to 10 and think about what we can do to fix this?

**Comment by jodh-intel** [comment]
@Pennyzct - do you have any calming emojis you can drop over here? :wink:

**Comment by gnawux** [comment]
@devimc,

Penny explained the snap-ci situation for me -- the snap-ci download the old kernel and will always fail for the PR. On the other hand, the 4.19.52 is a security update for a couple of CVEs, we can't wait for fixing all CIs. For a security update, we should merge it once it got enough `+1` and all the required CI passed.

On the other hand, one `+1` is from me. My thoughts on it: I don't think this patch broke any existing stuff
- Firstly, this is an arch specific update, and it won't affect other archs.
- Secondly, the aarch64 developer Penny confirmed it worked and didn't break more cases on either packaging tests and runtime CIs.

Finally, as James stated, the only failed aarch64 case in runtime repo is the chrony one, we should solve that soon.

Have a good weekend.

**Comment by Pennyzct** [comment]
Hi~ 
@jodh-intel  🍨🍧🍦 ice-cream is always the best option here.
the fault comes from my huge, obnoxious kernel patches😡🤬, people are asking for 4.19.52,  which is a security update for a couple of CVEs. So i raised this PR to do a little, tiny twist for the changing from 4.19.28 to 4.19.52.
@devimc FWIT, kernel version for snap stays in [v1.8.0-alpha0](https://github.com/kata-containers/packaging/blob/master/snap/snapcraft.yaml#L182
), which is still the old 4.19.28. Soooooo, o(TヘTo)
I think we could include eric's kernel update PR [runtime/1817](https://github.com/kata-containers/runtime/pull/1817) into a kata new version, maybe v1.8.0-alpha4. and update snap to this version v1.8.0-alpha4, and it will get fixed. ;)

**Comment by jodh-intel** [comment]
Thanks @Pennyzct - :ice_cream: sounds great, particularly with :cherries: and :chocolate_bar: on top ;)

Let's also add :musical_note:, a :fountain:, :sun_with_face:, :sunglasses: and some relaxing :fish: to watch :smile:

Your plan sounds good to me - wdyt @egernst?

**Comment by gnawux** [comment]
@bergwolf Do as what @Pennyzct said?

> I think we could include eric's kernel update PR runtime/1817 into a kata new version, maybe v1.8.0-alpha4. and update snap to this version v1.8.0-alpha4, and it will get fixed. ;)

**Comment by bergwolf** [comment]
@gnawux Yeah, now that all the kernel update PRs are merged, let's roll out a series of new releases.

**Pull request merged**